### PR TITLE
Remove link from CLI

### PR DESCRIPTION
**Pull request by rhatdan**
We have decided not to support using the --link param in the CLI.

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

File: buildah.go [hunk 1]
```diff
@@ -174,8 +167,7 @@ func OpenBuilder(store storage.Store, container string) (*Builder, error) {
 }
 
 // OpenBuilderByPath loads information about a build container given either a
```
Comment (nalind) [inline buildah.go]: Needs to remove the `either` here for the comment to read correctly.

File: cmd/buildah/config.go [hunk 2]
```diff
@@ -24,14 +24,10 @@ var (
 		},
 		cli.StringFlag{
 			Name:   "root",
-			Usage:  "root `directory` of the working container",
+			Usage:  "royot `directory` of the working container",
```
Comment (nalind) [inline cmd/buildah/config.go]: Typo?

**Comment by rhatdan** [comment]
Fixed.

File: cmd/buildah/addcopy.go [hunk 3]
```diff
@@ -38,20 +34,13 @@ func addAndCopyCmd(c *cli.Context, extractLocalArchives bool) error {
...
-	}
 	dest := ""
 	if c.IsSet("dest") {
 		dest = c.String("dest")
 	}
-	if name == "" && root == "" && link == "" {
+	if name == "" && root == "" {
 		if len(args) == 0 {
-			return fmt.Errorf("either a container name or --root or --link, or some combination, must be specified")
+			return fmt.Errorf("either a container name or --root, or some combination, must be specified")
```
Comment (TomSweeneyRedHat) [inline cmd/buildah/addcopy.go]: Are we sticking with container name or container id for our verbiage?  I'd prefer id, but would like to be consistent.  For this message and others similar to it, can we make it:

"either a container name or --root must be specified"

File: cmd/buildah/commit.go [hunk 4]
```diff
@@ -64,9 +56,9 @@ func commitCmd(c *cli.Context) error {
 	if !c.IsSet("disable-compression") || !c.Bool("disable-compression") {
 		compress = archive.Gzip
 	}
-	if name == "" && root == "" && link == "" {
+	if name == "" && root == "" {
 		if len(args) == 0 {
-			return fmt.Errorf("either a container name or --root or --link, or some combination, must be specified")
+			return fmt.Errorf("either a container name or --root, or some combination, must be specified")
```
Comment (TomSweeneyRedHat) [inline cmd/buildah/commit.go]: ditto 
"either a container name or --root must be specified"

File: cmd/buildah/config.go [hunk 5]
```diff
@@ -176,13 +172,9 @@ func configCmd(c *cli.Context) error {
...
 	root := c.String("root")
-	link := ""
-	if c.IsSet("link") {
-		link = c.String("link")
-	}
-	if name == "" && root == "" && link == "" {
+	if name == "" && root == "" {
 		if len(args) == 0 {
-			return fmt.Errorf("either a container name or --root or --link, or some combination, must be specified")
+			return fmt.Errorf("either a container name or --root, or some combination, must be specified")
```
Comment (TomSweeneyRedHat) [inline cmd/buildah/config.go]: Ditto
"either a container name or --root must be specified"

File: cmd/buildah/delete.go [hunk 6]
```diff
@@ -32,13 +28,9 @@ func deleteCmd(c *cli.Context) error {
...
 	root := c.String("root")
-	link := ""
-	if c.IsSet("link") {
-		link = c.String("link")
-	}
-	if name == "" && root == "" && link == "" {
+	if name == "" && root == "" {
 		if len(args) == 0 {
-			return fmt.Errorf("either a container name or --root or --link, or some combination, must be specified")
+			return fmt.Errorf("either a container name or --root, or some combination, must be specified")
```
Comment (TomSweeneyRedHat) [inline cmd/buildah/delete.go]: ditto 
"either a container name or --root must be specified"

File: cmd/buildah/addcopy.go (hunk 3 above)
Comment (nalind) [inline cmd/buildah/addcopy.go]: While they're different attributes (containers have exactly one ID, that's a hex representation of a 256-bit random number, while they can have zero or more names), the storage library handles name-to-ID lookups for us, so for the most part they're interchangeable.

File: cmd/buildah/mount.go [hunk 7]
```diff
@@ -32,16 +28,9 @@ func mountCmd(c *cli.Context) error {
...
-		link = c.String("link")
-		if link == "" {
-			return fmt.Errorf("link location can not be empty")
-		}
-	}
-	if name == "" && root == "" && link == "" {
+	if name == "" && root == "" {
 		if len(args) == 0 {
-			return fmt.Errorf("either a container name or --root or --link, or some combination, must be specified")
+			return fmt.Errorf("either a container name or --root, or some combination, must be specified")
```
Comment (TomSweeneyRedHat) [inline cmd/buildah/mount.go]: ditto
"either a container name or --root must be specified"

File: cmd/buildah/run.go [hunk 8]
```diff
@@ -62,9 +54,9 @@ func runCmd(c *cli.Context) error {
 	if c.IsSet("runtime") {
 		runtime = c.String("runtime")
 	}
-	if name == "" && root == "" && link == "" {
+	if name == "" && root == "" {
 		if len(args) == 0 {
-			return fmt.Errorf("either a container name or --root or --link, or some combination, must be specified")
+			return fmt.Errorf("either a container name or --root, or some combination, must be specified")
```
Comment (TomSweeneyRedHat) [inline cmd/buildah/run.go]: ditto
"either a container name or --root must be specified"

File: cmd/buildah/umount.go [hunk 9]
```diff
@@ -32,16 +28,9 @@ func umountCmd(c *cli.Context) error {
...
-		link = c.String("link")
-		if link == "" {
-			return fmt.Errorf("link location can not be empty")
-		}
-	}
-	if name == "" && root == "" && link == "" {
+	if name == "" && root == "" {
 		if len(args) == 0 {
-			return fmt.Errorf("either a container name or --root or --link, or some combination, must be specified")
+			return fmt.Errorf("either a container name or --root, or some combination, must be specified")
```
Comment (TomSweeneyRedHat) [inline cmd/buildah/umount.go]: ditto
"either a container name or --root must be specified"

File: delete.go [hunk 10]
```diff
@@ -2,19 +2,11 @@ package buildah
 
 import (
 	"fmt"
-	"os"
 )
 
 // Delete removes the working container.  The Builder object should not be used
```
Comment (TomSweeneyRedHat) [inline delete.go]: Should the comment say "The Container object" rather than "The Builder object" here?

File: cmd/buildah/addcopy.go (hunk 3 above)
Comment (TomSweeneyRedHat) [inline cmd/buildah/addcopy.go]: Thanks for the explanation.  I still think we should call it a container id in the documentation/calls/params and then in the help explain that a name will work too.  I'll head nod to whatever you and Dan this is best on this, but that's my $.02.

File: cmd/buildah/umount.go (hunk 9 above)
Comment (rhatdan) [inline cmd/buildah/umount.go]: Removed --root also and changed these to say 

container ID must be specified

File: cmd/buildah/addcopy.go (hunk 3 above)
Comment (rhatdan) [inline cmd/buildah/addcopy.go]: I am referring to them as "container ID"

File: cmd/buildah/commit.go (hunk 4 above)
Comment (rhatdan) [inline cmd/buildah/commit.go]: --root is gone.

File: delete.go (hunk 10 above)
Comment (rhatdan) [inline delete.go]: Lets discuss this one outside of this pull request.  I think Nalin is referring to the Buildah object in the code.

Comment (nalind) [inline delete.go]: Yes, though I wouldn't be opposed to changing the comment to reference it as a buildah.Builder object.

File: cmd/buildah/umount.go (hunk 9 above)
Comment (nalind) [inline cmd/buildah/umount.go]: Ah, that change is in #27.

**Comment by nalind** [comment]
Merging, to be continued in #27.

File: delete.go (hunk 10 above)
Comment (TomSweeneyRedHat) [inline delete.go]: That would be fine with me Nalin.

**Pull request merged**